purchase_order_partner_ref_mandatory/models/purchase_order.py

```diff
@@ -0,0 +1,10 @@
+# © 2022 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import  models, fields
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    partner_ref = fields.Char(required=True)
```

Comment:
It's technically a good option, but it's preferred to add `required` attribute for an existing field to the `form` view, in order to prevent latel problems (such as background processes that create purchase orders, that should fail if this field is not filled).